sequencer/src/bin/utils/ns_aggregator.rs

```diff
@@ -0,0 +1,100 @@
+use std::cmp::min;
+
+use anyhow::ensure;
+use clap::Parser;
+use espresso_types::{Header, NamespaceId, NamespaceProofQueryData};
+use futures::{future::try_join_all, stream::StreamExt};
+use sequencer::SequencerApiVersion;
+use surf_disco::Url;
+
+#[derive(Debug, Parser)]
+pub struct Options {
+    #[clap(short, long, default_value = "0")]
+    from_block: usize,
+
+    #[clap(short, long)]
+    to_block: Option<usize>,
+
+    #[clap(short, long)]
+    namespace: u64,
+
+    #[clap(short, long, default_value = "1")]
+    jobs: usize,
+
+    url: Url,
+}
+
+pub async fn run(opt: Options) -> anyhow::Result<()> {
+    let ns = NamespaceId::from(opt.namespace);
+    let client = surf_disco::Client::<hotshot_query_service::Error, SequencerApiVersion>::new(
+        opt.url.clone(),
+    );
+
+    let from = opt.from_block;
+    let to = match opt.to_block {
+        Some(to) => to,
+        None => client.get("status/block-height").send().await?,
+    };
+    ensure!(to >= from, "to-block < from-block");
+
+    let tasks = (0..opt.jobs).map(|i| {
+        let chunk_size = (to - from) / opt.jobs;
+        let start = from + i * chunk_size;
+        let end = min(start + chunk_size, to);
```

Comment:
`to` here in inclusive. Edge case: to = 1, from = 0, jobs=2 (for whatever reason). start = end = 0, block 1 never accounted for. no?

Reply:
Ah `to` was meant to be exclusive here, but I guess that could be confusing. I'll change it to be inclusive and fix this

Reply:
[a419fc2](https://github.com/EspressoSystems/espresso-network/pull/2839/commits/a419fc21c5eb4468b7d0d04028f234d6d3d84b2f)